veracruz-client/src/veracruz_client.rs

```diff
@@ -381,105 +384,98 @@ impl VeracruzClient {
-                        }
+                if let Some((received_session_id, received_data_vec)) = self.post_runtime_manager(enclave_session_id, &outgoing_data)? {
+                    enclave_session_id = received_session_id;
+                
+                    for received_data in received_data_vec {
+                        incoming_data_vec.push(received_data);
                     }
-                    None => (),
                 }
             }
 
             outgoing_data_vec.clear();
-            if incoming_data_vec.len() > 0 {
+
+            if !incoming_data_vec.is_empty() {
                 for incoming_data in &incoming_data_vec {
-                    let outgoing_data_option = self.process(incoming_data.to_vec())?;
-                    match outgoing_data_option {
-                        Some(outgoing_data) => {
-                            outgoing_data_vec.push(outgoing_data);
-                        }
-                        None => (),
+                    if let Some(outgoing_data) = self.process(incoming_data.to_vec())? {
+                        outgoing_data_vec.push(outgoing_data);
                     }
                 }
                 incoming_data_vec.clear();
             } else {
                 // try process with no data to see if it wants to send
-                let empty_vec = Vec::new();
-                let outgoing_data_option = self.process(empty_vec)?;
-                match outgoing_data_option {
-                    Some(outgoing_data) => outgoing_data_vec.push(outgoing_data),
-                    None => (),
+                if let Some(outgoing_data) = self.process(Vec::new())? {
+                    outgoing_data_vec.push(outgoing_data);
                 }
             }
-
-            let plaintext_data_option = self.get_data()?;
-            match plaintext_data_option {
-                Some(plaintext_data) => {
-                    self.remote_session_id = Some(enclave_session_id);
-                    return Ok(plaintext_data);
-                }
-                None => (),
-            }
+            
+            if let Some(plaintext_data) = self.get_data()? {
+                self.remote_session_id = Some(enclave_session_id);
+                return Ok(plaintext_data);
+            }    
         }
     }
 
     fn process(&mut self, input: Vec<u8>) -> Result<Option<Vec<u8>>, VeracruzClientError> {
         let mut ret_option = None;
-        let mut output: std::vec::Vec<u8> = std::vec::Vec::new();
-        if input.len() > 0 && (!self.tls_session.is_handshaking() || self.tls_session.wants_read())
-        {
+        let mut output: Vec<u8> = Vec::new();
+
+        if !input.is_empty() && (!self.tls_session.is_handshaking() || self.tls_session.wants_read()) {
             let mut slice = &input[..];
-            self.tls_session.read_tls(&mut slice)?;
 
+            self.tls_session.read_tls(&mut slice)?;
             self.tls_session.process_new_packets()?;
         }
+
```

Comment:
Is skipping line between blocks something that clippy recommends?

Reply:
No, that's something I prefer when reading code.